iroha/src/block_sync.rs

```diff
@@ -221,25 +220,19 @@ pub mod message {
                         return;
                     }
 
-                    if let Some(blocks) = block_sync.wsv.blocks_after(*hash).await {
-                        #[allow(clippy::cast_possible_truncation)]
-                        if let Some(blocks_batch) =
-                            blocks.chunks(block_sync.batch_size as usize).next()
-                        {
-                            if let Err(err) = Message::ShareBlocks(
-                                blocks_batch.to_vec(),
-                                block_sync.peer_id.clone(),
-                            )
-                            .send_to(peer)
-                            .await
-                            {
-                                iroha_logger::error!("Failed to send blocks: {:?}", err)
+                    match block_sync.wsv.blocks_after(*hash, block_sync.batch_size) {
+                        Ok(blocks) => {
+                            if !blocks.is_empty() {
```

Comment:
Could you move this predicate into match? Like this:
```rust
match block_sync.wsv.blocks_after(*hash, block_sync.batch_size) {
  Ok(blocks) if !blocks.is_empty() => { ... }
```

Reply:
I don't think it makes sense here

Reply:
That would remove 1 tab in terms of nesting, and also it will be easier to read

Reply:
You could also match after to just bare `Ok(_)` with empty hand

Reply:
I mean it should be combined with match as they are checking essentially different conditions.

Reply:
They would always check for differen logic (in essence), it is just for sepparating different cases without nesting. For easy reading of code

Reply:
Hmm seems too individual... To my mind, with nesting it seems clearer here.

Reply:
I mean, match is about FSMs, it is way clearer to read all the states linearly, scoping most of the variables and states.

I guess you can resolve this, as it is too individual.